File: php/src/Shared/OrderConstraint.php
<?php

namespace ImmutableDatamodel\Shared;

enum OrderConstraint
{
    case SHIPPING_JAPAN_ONLY;
    case DELIVERY_WEEKDAY;

    public function getViolationMessage(): string
    {
        return match($this) {
            OrderConstraint::SHIPPING_JAPAN_ONLY => 'This order is shipping only for Japan',
            OrderConstraint::DELIVERY_WEEKDAY => 'This order can deliver only on weekday',
        };
    }
}

File: golang/src/shared/order_constraint.go
package shared

type OrderConstraint int

const (
	SHIPPING_JAPAN_ONLY OrderConstraint = iota
	DELIVERY_WEEKDAY
)

var messages = map[OrderConstraint]string{
	SHIPPING_JAPAN_ONLY: "this order is shipping only for Japan",
	DELIVERY_WEEKDAY:    "this order can deliver only on weekday",
}

func (oc OrderConstraint) ViolationMessage() string {
	return messages[oc]
}
